.github/workflows/deploy-docs.yml

```diff
@@ -8,26 +8,95 @@ on:
     paths:
       - '.github/workflows/deploy-docs.yml'
       - 'docs/**'
+    # NOTE: The tag push event is not constrained by branch and path `on` conditions.
+    # For taking a snapshot of the docs state from the tagged commit,
+    # to a versioned subdirectory with dir name matching the git tag substring `v<MAJOR>.<MINOR>`:
+    tags:
+      - 'v[0-9]+.[0-9]+*'
 
-# Jobs will run shell commands from this subdirectory:
-defaults:
-  run:
-    working-directory: docs
+env:
+  # Default docs version to build and deploy
+  DOCS_VERSION: edge
+  GIT_USER: 'github-actions[bot]'
+  GIT_EMAIL: '41898282+github-actions[bot]@users.noreply.github.com'
 
 jobs:
+  add-version-to-docs:
+    if: startsWith(github.ref, 'refs/tags/')
+    name: 'Handle `versions.json` in `gh-pages` branch'
+    runs-on: ubuntu-20.04
+    steps:
+      - name: 'Checkout the default branch (master)'
```

Comment:
Would remove `(master)` as default could change ;)

Reply:
Yes, but you could change that here too. It was just to provide some context. It's actually not entirely correct. I've learned that the checkout is checking out the commit for the triggered event only. Tags don't have any relation to a branch either, I thought I had fixed this name and pushed a change (probably have due to the "outdated" status on this message.